src/js/common/schemaform/widgets/SelectArrayItemsWidget.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import get from '../../utils/data-utils/get';
+import set from '../../utils/data-utils/set';
+
+// TODO: Safety checks for `selected` callback and `label` element
+
+export default class SelectArrayItemsWidget extends React.Component {
+  onChange = (index, checked) => {
+    const items = set(`[${index}].${this.props.options.selectedPropName}`, checked, this.props.value);
+    this.props.onChange(items);
+  }
+
+  render() {
+    const {
+      value: items,
+      id,
+      options,
+      required
+    } = this.props;
+    const { label: Label, selectedPropName, disabled } = options;
+
+    // TODO: Figure out how to handle required fields
```

Comment:
This should mostly be handled by StringField, if you mark the field as required as usual.